apps/central-scan/frontend/src/screens/admin_actions_screen.test.tsx

```diff
@@ -95,108 +90,72 @@ test('clicking "Save Backup" shows progress', async () => {
-  const unconfigureButton = screen.getByText(
-    'Delete Election Data from VxCentralScan'
-  );
-  unconfigureButton.click();
-  screen.getByText('Delete all election data?');
+  expect(
+    screen.getButton('Delete Election Data from VxCentralScan')
+  ).toBeDisabled();
 
-  // Clicking the disabled "Delete Ballot Data" button should bring up a confirmation modal
-  const deleteBallotsButton = screen.getByText('Delete Ballot Data');
-  deleteBallotsButton.click();
-  screen.getByText('Delete All Scanned Ballot Data?');
+  expect(screen.getButton('Delete Ballot Data')).toBeDisabled();
 });
 
-test('clicking "Delete Election Data from VxCentralScan" shows progress', async () => {
-  const unconfigureServer = jest.fn();
-  renderScreen({ unconfigureServer, canUnconfigure: true });
+test('clicking "Delete Election Data from VxCentralScan" calls backend', async () => {
+  mockApiClient.getMarkThresholdOverrides.expectCallWith().resolves(null);
 
-  const { promise, resolve } = deferred<void>();
-  unconfigureServer.mockReturnValueOnce(promise);
+  const history = createMemoryHistory({ initialEntries: ['/admin'] });
+  renderScreen({ canUnconfigure: true }, history);
 
-  // Click to reset.
-  expect(unconfigureServer).not.toHaveBeenCalled();
-  const resetButton = screen.getByText(
-    'Delete Election Data from VxCentralScan'
-  );
-  resetButton.click();
+  // initial button
+  userEvent.click(screen.getButton('Delete Election Data from VxCentralScan'));
 
-  // Confirm reset.
-  expect(unconfigureServer).not.toHaveBeenCalled();
+  // first confirmation
   screen.getByText('Delete all election data?');
-  const confirmResetButton = await waitFor(() =>
-    screen.getByText('Yes, Delete Election Data')
-  );
-  confirmResetButton.click();
+  userEvent.click(await screen.findButton('Yes, Delete Election Data'));
+
+  // second confirmation
+  mockApiClient.unconfigure
+    .expectCallWith({ ignoreBackupRequirement: false })
+    .resolves(ok());
   screen.getByText('Are you sure?');
-  const doubleConfirmResetButton = await waitFor(() =>
-    screen.getByText('I am sure. Delete all election data.')
+  userEvent.click(
+    await screen.findButton('I am sure. Delete all election data.')
   );
-  doubleConfirmResetButton.click();
-  expect(unconfigureServer).toHaveBeenCalledTimes(1);
 
-  // Verify progress message is shown.
-  await waitFor(() => screen.getByText('Deleting election data'));
+  // progress message
+  await screen.findByText('Deleting election data');
 
-  // Trigger reset finished, verify back to initial screen.
-  resolve();
-  await waitFor(() => !screen.getByText('Deleting election data'));
+  // we are redirected to the dashboard
+  expect(history.location.pathname).toEqual('/');
 });
 
-test('clicking "Delete Ballot Data" shows progress', async () => {
-  const zeroData = jest.fn();
-  renderScreen({ zeroData, hasBatches: true, canUnconfigure: true });
+test('clicking "Delete Ballot Data" calls progress', async () => {
```

Comment:
is this meant to say "calls backend" not "calls progress"?

Reply:
4d79f44f2